homework/content.js

```diff
@@ -1,37 +1,64 @@
 'use strict';
 
-(function() {
-  var _wrapper = document.querySelector('#note-content-wrapper');
+(function(exports) {
 
-  function start() {
-    window.addEventListener('note-open', function(event) {
-      var note = event.detail;
-      resetWrapper();
-      drawNote(note);
-    });
+  /**
+   * Creates an instance of ContentManager.
+   *
+   * @constructor
+   * @this {ContentManager}
+   */
+  var ContentManager = function() {
+    this._wrapper = document.querySelector('#note-content-wrapper');
   }
 
-  function resetWrapper() {
-    _wrapper.innerHTML = '';
-  }
+  ContentManager.prototype = {
 
-  function drawNote(note) {
-    var title = note.title;
-    var h = document.createElement('h2');
-    h.textContent = title;
-    var passages = note.passages;
-    var buff = document.createDocumentFragment();
-    passages.forEach(function(passage) {
-      var p = document.createElement('p');
-      p.classList.add('note-passage');
-      p.textContent = passage;
-      buff.appendChild(p);
-    });
-    _wrapper.appendChild(h);
-    _wrapper.appendChild(buff);
-  }
+    /**
+     * Binding the 'note-open' event to the '#note-content-wrapper' node.
+     * This method should be called only once to initialize the ContentManager.
+     *
+     * @this {ContentManager}
+     */
+    start() {
+      window.addEventListener('note-open', (function(event) {
```

Comment:
It's a convention to write listening code in `handleEvent` after binding that as `window.addEventListener('note-open', this)`:

    window.addEventListener('note-open', this);
    //....

    handleEvent(event) {
      switch (event.type) {
        case 'note-open':
            this.onNoteOpen(event.detail);
            break;
      }
    }

    onNoteOpen(note) {
      this.resetWrapper();
      this.drawNote(note);
    }

To centralize the event receiving and dispatching code is to prevent debugging in a file with lots of registering and handling code scatter about the whole component.